### PR TITLE
change to platform-python #48

### PR DESCRIPTION
**Pull request by flybyray**
only the platform-python ensures that the python3-selinux binding would be available.
newer python from modules need to reference the selinux bindings from platform-python

https://github.com/pycontribs/selinux/issues/48

### CODE REVIEW

File: selinux/__init__.py [hunk 1]
```diff
@@ -73,12 +73,10 @@ def get_system_sitepackages():
         """Get sitepackage locations from system python"""
         # Do not ever use sys.executable here
         # See https://github.com/pycontribs/selinux/issues/17 for details
-        system_python = "/usr/bin/python%s" % platform.python_version_tuple()[0]
-
         system_sitepackages = json.loads(
             subprocess.check_output(
                 [
-                    system_python,
+                    "/usr/libexec/platform-python",
```
Comment (ssbarnea) [inline selinux/__init__.py:79]: That will not work correctly. Keep in mind that you can have multiple versions of python installed at the same time. You change means that it will stop working for all cases where current python version is different than the system one.

**Review by ssbarnea — CHANGES_REQUESTED** [review]

File: selinux/__init__.py (hunk 1 above)
Comment (flybyray) [reply selinux/__init__.py:79]: ok i understand your point for old systems. but the current implementation will not work for redhat family systems major version 8. The selinux bindings will not be available for any other python installed. It is only available for the 
/usr/libexec/platform-python

is there any supported way to override this  selection of a system python?
https://github.com/pycontribs/selinux/blob/321d66619ed59bec00a9dc6216ce417d7befec41/selinux/__init__.py#L76
it is just wrong and if i inject the correct version or use the /usr/libexec/platform-python ( which will be selected as a fallback in ansible ) than ansible installed in a virtualenv with newer python will work.

Comment (flybyray) [reply selinux/__init__.py:79]: @ssbarnea maybe this minimal example will show you what i mean

start a suitable docker/podman container for example like this:
`docker run -it --rm --user root registry.access.redhat.com/ubi8/python-39 bash`

install requirements and run just an ansible command:

```shell
dnf install -y selinux-policy
pip install selinux ansible
ansible -m ping localhost

```
output:

```text
Traceback (most recent call last):
  File "/opt/app-root/bin/ansible", line 104, in <module>
    mycli = getattr(__import__("ansible.cli.%s" % sub, fromlist=[myclass]), myclass)
  File "/opt/app-root/lib64/python3.9/site-packages/ansible/cli/__init__.py", line 24, in <module>
    from ansible.parsing.dataloader import DataLoader
  File "/opt/app-root/lib64/python3.9/site-packages/ansible/parsing/dataloader.py", line 17, in <module>
    from ansible.module_utils.basic import is_executable
  File "/opt/app-root/lib64/python3.9/site-packages/ansible/module_utils/basic.py", line 144, in <module>
    from ansible.module_utils.common.process import get_bin_path
  File "/opt/app-root/lib64/python3.9/site-packages/ansible/module_utils/common/process.py", line 9, in <module>
    from ansible.module_utils.common.file import is_executable
  File "/opt/app-root/lib64/python3.9/site-packages/ansible/module_utils/common/file.py", line 24, in <module>
    import selinux
  File "/opt/app-root/lib64/python3.9/site-packages/selinux/__init__.py", line 104, in <module>
    check_system_sitepackages()
  File "/opt/app-root/lib64/python3.9/site-packages/selinux/__init__.py", line 100, in check_system_sitepackages
    raise Exception(
Exception: Failed to detect selinux python bindings at ['/usr/local/lib64/python3.9/site-packages', '/usr/local/lib/python3.9/site-packages', '/usr/lib64/python3.9/site-packages', '/usr/lib/python3.9/site-packages']

```

Comment (ssbarnea) [reply selinux/__init__.py:79]: What happens if you skip installing selinux using pip? I bet you get a similar error. This package will no install python-selinux bindings for you! You still need to install them for it to work.

Comment (flybyray) [reply selinux/__init__.py:79]: Better not bet. Otherwise I bet you did not read the stacktrace ( not even the last line ). 😃

The last line from stacktrace tells yoy that the error is thrown by selinux!
If it is not installed> how the "similar error" can be thrown?
From stacktrace you may notice an important path [file.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/common/file.py#L27) 

```
(app-root) ansible -m ping localhost
[WARNING]: No inventory was parsed, only implicit localhost is available
localhost | SUCCESS => {
    "changed": false,
    "ping": "pong"
}
```

The problem is the current implementation expects python installed system wide.

I think we need to find some other solution.

Currently ansible's stable release version is 2.11.2 from 22 June 2021. When 2.12 is released, python 3.6 is not longer supported. But 3.6 is a default python3 for Centos/RHEL 7/8 . A lot of people will install python 3.8+ maybe not system wide or in not common places system wide.

I try to figure out what would be a better solution. Maybe you have an idea?
For me the next obvious solution would be to use the python from the current env. At best the [`ansible_python_interpreter`](https://docs.ansible.com/ansible/latest/reference_appendices/interpreter_discovery.html).

Comment (ssbarnea) [reply selinux/__init__.py:79]: Let me rephrase this: if ansible works fine inside a virtualenv created with --system-packages but it fails inside a virtualenv created w/o them and with this selinux, is a bug on this module.

Comment (flybyray) [reply selinux/__init__.py:79]: ok thanks for that rephrase. i guess it is a bug on this module than. in this example centos based python image a virtualenv w/o system-pacakges is used by default:

```
(app-root) cat $VIRTUAL_ENV/pyvenv.cfg
home = /usr/bin
include-system-site-packages = false
version = 3.9.2
```
